x/dymns/keeper/alias.go

```diff
@@ -0,0 +1,178 @@
+	"github.com/dymensionxyz/gerr-cosmos/gerrc"
+)
+
+// GetRollAppIdByAlias returns the RollApp-Id by the alias.
+func (k Keeper) GetRollAppIdByAlias(ctx sdk.Context, alias string) (rollAppId string, found bool) {
+	defer func() {
+		found = rollAppId != ""
+	}()
+
+	store := ctx.KVStore(k.storeKey)
+	key := dymnstypes.AliasToRollAppIdRvlKey(alias)
+	bz := store.Get(key)
+	if bz != nil {
+		rollAppId = string(bz)
+		return
+	}
+
+	for rid, data := range mockRollAppsData {
+		if data.alias == alias {
+			rollAppId = rid
+			return
+		}
+	}
+
+	return
+}
+
+// GetAliasByRollAppId returns the alias by the RollApp-Id.
+func (k Keeper) GetAliasByRollAppId(ctx sdk.Context, chainId string) (alias string, found bool) {
+	// TODO DymNS: support returns multiple aliases
+	if !k.IsRollAppId(ctx, chainId) {
+		return
+	}
+
+	defer func() {
+		found = alias != ""
+	}()
+
+	store := ctx.KVStore(k.storeKey)
+	key := dymnstypes.RollAppIdToAliasesKey(chainId)
+	bz := store.Get(key)
+	if bz != nil {
+		var multipleAliases dymnstypes.MultipleAliases
+		k.cdc.MustUnmarshal(bz, &multipleAliases)
+		if len(multipleAliases.Aliases) > 0 {
+			alias = multipleAliases.Aliases[0]
+			return
+		}
+	}
+
+	if data, ok := mockRollAppsData[chainId]; ok {
+		alias = data.alias
+		return
+	}
+
+	return
+}
+
+// SetAliasForRollAppId assigns the usage of an alias to a RollApp.
+func (k Keeper) SetAliasForRollAppId(ctx sdk.Context, rollAppId, alias string) error {
+	if !k.IsRollAppId(ctx, rollAppId) {
+		return errorsmod.Wrapf(gerrc.ErrInvalidArgument, "not a RollApp chain-id: %s", rollAppId)
+	}
+
+	if alias == "" {
+		return errorsmod.Wrapf(gerrc.ErrInvalidArgument, "alias can not be empty")
+	}
+
+	if !dymnsutils.IsValidAlias(alias) {
+		return errorsmod.Wrapf(gerrc.ErrInvalidArgument, "invalid alias: %s", alias)
+	}
+
+	store := ctx.KVStore(k.storeKey)
+	keyR2A := dymnstypes.RollAppIdToAliasesKey(rollAppId)
+	keyA2R := dymnstypes.AliasToRollAppIdRvlKey(alias)
+
+	if bz := store.Get(keyA2R); bz != nil {
+		return errorsmod.Wrapf(gerrc.ErrAlreadyExists, "alias currently being in used by: %s", string(bz))
+	}
+
+	var multipleAliases dymnstypes.MultipleAliases
+	if bz := store.Get(keyR2A); bz != nil {
+		k.cdc.MustUnmarshal(bz, &multipleAliases)
+	}
+	multipleAliases.Aliases = append(multipleAliases.Aliases, alias)
+
+	store.Set(keyR2A, k.cdc.MustMarshal(&multipleAliases))
+	store.Set(keyA2R, []byte(rollAppId))
+
+	return nil
+}
+
+// RemoveAliasFromRollAppId removes the linking of an existing alias from a RollApp.
+func (k Keeper) RemoveAliasFromRollAppId(ctx sdk.Context, rollAppId, alias string) error {
+	if !k.IsRollAppId(ctx, rollAppId) {
+		return errorsmod.Wrapf(gerrc.ErrInvalidArgument, "not a RollApp chain-id: %s", rollAppId)
+	}
+
+	if alias == "" {
+		return errorsmod.Wrapf(gerrc.ErrInvalidArgument, "alias can not be empty")
+	}
+
+	if !dymnsutils.IsValidAlias(alias) {
+		return errorsmod.Wrapf(gerrc.ErrInvalidArgument, "invalid alias: %s", alias)
+	}
```

Comment:
remove? no thanks, I tried to make sure no surprise call

Reply:
@VictorTrustyDev no, not removed. Extracted into a method, because a lot of methods have this same part, so it can be moved into a new method.

DRY: [Don't Repeat Yourself](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

Reply:
yo my brain broken that time lol

Reply:
no worries, mine too

Reply:
This piece of code repeated 3 times.
It is possible to have a method for it, but the problem is the business logic is not able to have a good enough name for it, kinda: `is roll app exists and alias valid`, that so weird so I decided to not extract it.
No thanks.